### PR TITLE
Improve compatibility with IDEs plus mention lombok plugin requirement

### PR DESCRIPTION
**Pull request by balazstbb**
- build helper plugin added to attache OpenAPI generated sources
- gitignore extended with Theia IDE support
- documentation (README.md) IDE section extended the information lombok plugin is required

### CODE REVIEW

**Review by torrespro — CHANGES_REQUESTED** [review]
Balazs could you review why the build is failing? looks like the SSDK is not in repo yet? or it was not...

**Comment by balazstbb** [comment]
hm, interesting.... this should pull dependencies from backbase repo and that is private.... I have not changed anything with it... I do not know how is suppose to connect to backbase maven repo

**Comment by torrespro** [comment]
so the other builds worked, with the same settings and Maven username and pass, so it must be something else

File: .gitignore [hunk 1]
```diff
@@ -34,6 +34,13 @@ build/
 .vscode/
 node_modules/
 
+### Theia ###
+.theia
+.classpath
+.factorypath
+.project
+.settings
+
```
Comment (pa314159) [inline .gitignore:43]: All these dots can be specified by the following patterns
```
.*
!.git?*
!.mvn/

```

File: services/review/pom.xml [hunk 2]
```diff
@@ -235,6 +235,24 @@
...
             </plugin>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>build-helper-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>add-source</id>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>add-source</goal>
```
Comment (pa314159) [inline services/review/pom.xml:246]: Boat already supports `addCompileSourceRoot` which defaults to true, isn't it enough?

**Review by pa314159 — CHANGES_REQUESTED** [review]

File: services/review/pom.xml (hunk 2 above)
Comment (balazstbb) [reply services/review/pom.xml:246]: Did not work, I tried with Eclipse and with Theia (VSCode), unless I attached the additional sources with build-helper plugin I have compile errors in my IDE (the code compiles of course, but the IDE can not pick it up)

**Pull request closed without merging**